indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -84,14 +89,20 @@ public WorkerTaskMonitor(
   public void start()
   {
     try {
+      // restore restorable tasks
+      final List<Pair<Task, ListenableFuture<TaskStatus>>> restored = taskRunner.restore();
+      for (Pair<Task, ListenableFuture<TaskStatus>> pair : restored) {
+        submitTaskRunnable(pair.lhs, pair.rhs);
```

Comment:
What happens if the config changes and the total resource pool has gone down? I think this will block when trying to add the task to the queue.